### PR TITLE
octopus: mgr/dashboard: allow preserving OSD IDs when deleting OSDs

### PR DESCRIPTION
**Pull request by callithea**
backport tracker: https://tracker.ceph.com/issues/46173

---

backport of https://github.com/ceph/ceph/pull/35242
parent tracker: https://tracker.ceph.com/issues/38234

this backport was staged using ceph-backport.sh version 15.1.1.389
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by callithea** [comment]
`make check` failure might be related to changes in master of the [osd-list.component.ts](https://github.com/bk201/ceph/commit/3069600461c080e873ac84906adba24835a3f556#diff-af8bd92dca93b52f51e8c247482bd559) due to the angular 9 update, which would require more adaptations to be made in order to get it working for octopus..?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts [hunk 1]
```diff
@@ -54,6 +56,8 @@ export class OsdListComponent extends ListWithDetails implements OnInit {
   reweightBodyTpl: TemplateRef<any>;
   @ViewChild('safeToDestroyBodyTpl', { static: false })
   safeToDestroyBodyTpl: TemplateRef<any>;
+  @ViewChild('deleteOsdExtraTpl')
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts:59]: In Angular 8 we need to specify `static` on `@ViewChild`:
```suggestion
  @ViewChild('deleteOsdExtraTpl', { static: false })
```

Comment (callithea) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts:59]: @tspmelo: Yes, that's why I was wondering if there was a plan to update octopus to angular 9 as well or if I should rather manually fix this line in order to stick with angular 8 (will fix this line then as suggested, thanks for taking a look :) ).

**Comment by callithea** [comment]
jenkins test make check

**Comment by bk201** [comment]
jenkins test dashboard

**Comment by yuriw** [comment]
[wip-yuri7-testing-2020-07-06-1754-octopus](https://trello.com/c/VGUTyb8R/952-wip-yuri7-testing-2020-07-06-1754-octopus)

**Pull request merged**